internal/k8s/reconciler/factory.go

```diff
@@ -10,6 +10,7 @@ import (
 	apigwv1alpha1 "github.com/hashicorp/consul-api-gateway/pkg/apis/v1alpha1"
 )
 
+// TODO Remove Factory as it's no longer necessarygithub
```

Comment:
`Factory` is no longer really necessary and will be removed in the final cleanup pass